### DISCUSSION TITLE
Not able to ssh to AWS instances using github actions

### DISCUSSION THREAD

**Discussion by aniketm1996** [community/community#65352] [Actions]
### Select Topic Area

General

### Body

I am experiencing difficulty connecting to AWS instances using Github Actions . However, when I modified the security group to allow SSH access to all, the connection started working again.

- name: SSH into EC2 and run commands
      uses: appleboy/ssh-action@master
      with:
        host: 48.32.123.34
        username: ec2-user
        key: ${{ secrets.SSH_PRIVATE_KEY }}
        script: |
          # Commands to run on the EC2 instance
          echo "Hello, EC2!"
          ls

**Comment by arafajs** [comment]
The reason why you are not able to SSH to AWS instances using GitHub Actions is because the security group does not allow SSH access from your GitHub Actions runner. When you modified the security group to allow SSH access to all, the connection started working again because your GitHub Actions runner is now allowed to SSH to the EC2 instance.

To fix this, you need to make sure that the security group for your EC2 instance allows SSH access from your GitHub Actions runner. You can do this by following these steps:

1. Go to the AWS Management Console and select the EC2 service.
2. In the left navigation pane, select Security Groups.
3. Select the security group for your EC2 instance.
4. In the Inbound Rules section, click Edit.
5. In the Add Rule dialog box, select SSH from the Type drop-down list.
6. In the Source field, enter the IP address or CIDR range of your GitHub Actions runner.
7. Click Add.

**Comment by aniketm1996** [comment]
@arafajs 

Where we should get CIDR range or IP address because Github actions runner ip's are many.
Can u please get me the exact CIDR range or Ip addresses?.

Thank you in advance.

**Comment by aniketm1996** [comment]
@arafajs Any idea?

**Comment by abriginets** [comment]
@aniketm1996 try https://api.github.com/meta
